File: src/LeetCode/Algorithms/Java/Move_Zeroes.java
package LeetCode.Algorithms.Java;

// https://leetcode.com/problems/move-zeroes/
public class Move_Zeroes {
	public static void main(String[] args) {
		int[] nums = {0, 1, 0, 3, 12};
		
		moveZeroes(nums);
	}
	
	public static void moveZeroes(int[] nums) {
		int index = 0;
		
		for (int n : nums) {
			if (n != 0) {
				nums[index++] = n;
			}
		}
		
		for (int i = index; i < nums.length; i++) {
			nums[i] = 0;
		}
	}
}

File: src/LeetCode/Algorithms/JavaScript/Move_Zeroes.js
/**
 * https://leetcode.com/problems/move-zeroes/
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
	let index = 0;
	
	for (let n of nums) {
		if (n !== 0) {
			nums[index++] = n;
		}
	}
	
	for (let i = index; i < nums.length; i++) {
		nums[i] = 0;
	}
};

let nums = [0, 1, 0, 3, 12];

console.log(moveZeroes(nums));